src/diffusers/models/vq_paella.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Union
+
+import torch
+import torch.nn as nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from .modeling_utils import ModelMixin
+from .resnet import MixingResidualBlock
+from .vae import DecoderOutput, VectorQuantizer
+from .vq_model import VQEncoderOutput
+
+
+class VQModelPaella(ModelMixin, ConfigMixin):
```

Comment:
Can we try to **not** create a new `Paella` class here and instead add the necessary changes to the `VQModel`: https://github.com/huggingface/diffusers/blob/38efac9f61584ca58203cad94a3c419a19e8684e/src/diffusers/models/vq_model.py#L39

I think all we need is to define some new blocks and add a `input_type="pixel_shuffle"` config flag for the input / output